students/perrymant/code-examples/cache/src/CacheTest.java

```diff
@@ -0,0 +1,40 @@
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+/*
+can we create tests to ensure we're not breaking anything please?
+This should give you some practice in dependency injection too,
+as you'll have to give the class a DB implementation when testing.
+*/
+@RunWith(MockitoJUnitRunner.class)
+public class CacheTest {
+
+    @Mock
+    DB db;
```

Comment:
We don't need mockito here. We're not using it presently anyway, so let's not try and confuse things with having to understand Mockito's syntax.

Can we create a DB implementation for the test instead?
Then you can decide, in your tests, what and when it should return certain data.